Use "Completed Order" component in Profile page

efalayi opened the pull request:
Resolves [#3018](https://github.com/reactioncommerce/reaction/issues/3018)
Displayed orders list on user profile page uses the old completed order Blaze template  but we should reuse the Completed Order component that was converted to React and made marketplace-aware.

###### Fix
- Create a template for userOrdersList to manage list display
- Create an helper function for userOrdersList template that return the `CompletedOrder` react component
- Create a container component for userOrdersList
- Fix CSS style errors
- Update profile template with new template

###### Test
```
Login as admin
Create an order
Navigate to user profile page
Scroll to displayed user order
Observe that displayed list matches the completedOrder template
```

zenweasel commented:
@efalayi As I've mentioned before, when we want Ryan to review something let's add the `design-review` label, not add him as a reviewer.

zenweasel commented:
So you will need to make some adjustments to the completed order component because some of those elements don't make sense within the context of the profile page, e.g. the giant "Thank you".

![account_profile](https://user-images.githubusercontent.com/72819/31591598-c264954a-b253-11e7-8532-1a5e39ea027c.png)

zenweasel commented on imports/plugins/core/accounts/client/templates/profile/userOrdersList.js:
```diff
@@ -0,0 +1,82 @@
+import CompletedOrder from "/imports/plugins/core/checkout/client/components/completedOrder";
+
+/**
+ * @method handleDisplayMedia
```
Isn't there a way we could be rendering the Completed Order container rather than duplicating all this code?

zenweasel commented on imports/plugins/core/accounts/client/templates/profile/userOrdersList.html:
```diff
@@ -0,0 +1,11 @@
+      <span data-i18n="cartCompleted.noOrdersFound">No orders found.</span>
+    </div>
+  {{/each}}
+</template>
```
Missing CR/LF

rymorgan commented:
Two things: 

1. Let's tweak the layout so that the items and cart portion stretch across the full box
2. Also, on the user profile, the title should change from 'Your cart' to 'Order summary'

![complete orders view - user profile](https://user-images.githubusercontent.com/20409254/31632187-e55c1c18-b270-11e7-8235-7f44ff1a38e8.png)

efalayi commented:
@ryan should the tweak in layout be specific to just the user profile page?

rymorgan commented:
@efalayi Yep, only on the user profile page.

efalayi commented:
@ryan I just observed that the layout above is different from the one here: https://scene.zeplin.io/project/58af8900737c137280322bd6/screen/59b7272087fcafc969870826. Do we still intend to use the layout on zeplin?

rymorgan commented:
@efalayi The layout in Zeplin is where we are going in the future w/ a refreshed design of our user profile views. The issue we are working in this PR is using the component consistently. So use the designs I provided here vs the Zeplin ones, which we will tackle when we work on the rest of the user profile updates.

efalayi commented:
@zenweasel @rymorgan I've implemented all requested changes on this PR.

rymorgan commented:
@efalayi I'm seeing some formatting problems and after that, this PR should be good form my perspective. 

Need to vertically center align the copy in the rows for 'Your items'. 

![account_profile](https://user-images.githubusercontent.com/20409254/31736730-61ed2c92-b3fa-11e7-9aee-3f24679df051.png)

Need to fix casing 
![account_profile-2](https://user-images.githubusercontent.com/20409254/31736825-a35e47b0-b3fa-11e7-81b0-28953a0f32f1.png)


On mobile - 
You can reduce the padding by a half. 
Stack shipping address and payment method. 
Make sure the containers are all centered. (email/password, your orders, ad address) and not getting cut off. 

![iphone_x_-_ios_11_0](https://user-images.githubusercontent.com/20409254/31736966-05fe3c5e-b3fb-11e7-8ebe-0662a73c47ef.png)

efalayi commented:
@rymorgan I've updated the PR with the changes specified.

rymorgan commented:
A few of my comments didn't get addressed yet. 

Match the spacing between outer box and inner box. The Your orders box should match the address box. So, we need to reduce the spacing to match.

<img width="417" alt="iphone_x_-_ios_11_0_and_inbox_ _morgan_ry_gmail_com" src="https://user-images.githubusercontent.com/20409254/31785341-be0555bc-b4b9-11e7-97cf-b093ead3ff73.png">

I know this wasn't in the original ticket, but while were on this page, I'd like to fix the casing issue on the address book title. Should read 'Address book' 

![account_profile-4](https://user-images.githubusercontent.com/20409254/31785401-ee7782f6-b4b9-11e7-934e-61736597a842.png)

I also noticed two other spacing issues see below: 

The spacing is inconsistent between all of the boxes. All the boxes w/ title should have even spacing between them. The Your items box has more space both above and below than it should. This is happening on mobile and desktop. 

![account_profile-5](https://user-images.githubusercontent.com/20409254/31785714-ed1a7ab6-b4ba-11e7-83db-feda23f817ca.png)

There's an awkward unevenness between the shipping and payment boxes on desktop. 

![account_profile-3](https://user-images.githubusercontent.com/20409254/31785560-67a1d6cc-b4ba-11e7-8f46-dac2e076d289.png)

efalayi commented:
@zenweasel I created an `OrdersList` component as you suggested and refactored the code. @rymorgan the css issues have also been fixed.

zenweasel commented:
@rymorgan Are we will saying "Your cart" in the completed order page? Should we just say "Your items"?

<img width="395" alt="cart_completed" src="https://user-images.githubusercontent.com/72819/31960102-886701e4-b8ab-11e7-864a-ce1ba9007253.png">

zenweasel commented:
@efalayi I spoke with @rymorgan and we agreed we should make it "Your items" (rather than You cart) on the completed order page as well as the profile page. Also we want to add Order ID on the Account version. Ryan has updated the comp.

rymorgan commented:
Here's the profile. 

![complete orders view - user profile](https://user-images.githubusercontent.com/20409254/31964936-8423589e-b8ba-11e7-836d-1491c83640eb.png)

efalayi commented:
I'll get to it and update the PR asap.

efalayi commented:
@zenweasel @rymorgan after making the change, this is what completed order page looks like:

<img width="1280" alt="completed-order-page" src="https://user-images.githubusercontent.com/28438068/31971759-1f9edd8e-b915-11e7-9bac-4a59b87252ba.png">

`Your items` is repeated twice. I'm not sure what the first does though.

rymorgan commented:
![31971759-1f9edd8e-b915-11e7-9bac-4a59b87252ba_png__2560x1450_](https://user-images.githubusercontent.com/20409254/31974320-e96ddf10-b8df-11e7-84e8-123f30eb4c99.png)

efalayi commented:
@zenweasel I just updated the PR with the changes. I noticed that no message was displayed if the user has no orders (that's now fixed in the update).

zenweasel commented on imports/plugins/core/accounts/client/templates/profile/userOrdersList.js:
```diff
@@ -0,0 +1,46 @@
+ * @since 1.5.1
+ * @return {Array} - an array of orders
+ */
+function getOrders() {
```
I'm confused as to why we are doing anything at all in the template. If we still need the template at all the only thing it should be doing is rendering the component.

zenweasel commented on imports/plugins/core/accounts/client/templates/profile/userOrdersList.html:
```diff
@@ -0,0 +1,11 @@
+      {{> React completedOrders }}
+    </div> 
+  {{else}}
+    <div class="alert alert-info">
```
The component should be handling both the found and not found cases.

efalayi replied:
Okay. I'll create functions to render found and not found cases in the component.

zenweasel commented on imports/plugins/core/checkout/client/components/completedOrder.js:
```diff
@@ -27,8 +28,15 @@ const CompletedOrder = ({ order, orderId, shops, orderSummary, paymentMethods, h
-  return (
-    <div className="container order-completed">
+
+  let headerText;
```
This component should not need to be aware of what route it is in. It should be handled by passing in a prop. Components should be as dumb as possible.

zenweasel commented on imports/plugins/core/checkout/client/components/completedShopOrders.js:
```diff
@@ -10,16 +9,24 @@ import CompletedOrderItem from "./completedOrderItem";
  * @property {String} shopName - The name of the shop
  * @property {Array} items - an array of individual items for this shop
  * @property {Function} handleDisplayMedia - A function for displaying product images
+ * @property {boolean} isProfilePage - Checks if current page is Profile Page
```
Consistent casing for Javascript types

zenweasel commented on imports/plugins/core/checkout/client/components/completedShopOrders.js:
```diff
@@ -10,16 +9,24 @@ import CompletedOrderItem from "./completedOrderItem";
         <div className="store-detail-box">
           <span className="order-details-store-title">{shopName}</span>
-          <span className="order-details-shipping-name">{shippingMethod.carrier} - {shippingMethod.label}</span>
+          <span className="order-details-shipping-name">{shippingName}</span>
```
Why was this changed?

efalayi replied:
`shippingName` varies depending on whether the current page is profile page. It is defined in  line 16.

zenweasel commented on imports/plugins/core/checkout/client/components/shopOrderSummary.js:
```diff
@@ -6,9 +6,11 @@ import { Components, registerComponent } from "@reactioncommerce/reaction-compon
  * @summary Displays the order summary for each shop
  * @param {Object} props - React PropTypes
  * @property {Object} shopSummary - An object representing the summary information for this Shop
+ * @property {boolean} isProfilePage - Checks if current page is Profile Page
```
Consistent casing

zenweasel commented:
Still not exactly right on mobile devices

<img width="424" alt="iphone_x_-_ios_11_0_and_applications_and_cart_completed" src="https://user-images.githubusercontent.com/72819/32241510-91961bce-be2d-11e7-83a2-81a951ae1230.png">

zenweasel commented:
Seems like this order break could be delineated better. Really gets lost on mobile.

<img width="885" alt="account_profile" src="https://user-images.githubusercontent.com/72819/32241649-fb1110fe-be2d-11e7-8c7f-5d3ae40bfd15.png">

rymorgan commented:
@zenweasel Agree, let's add a rule between different orders. Two things: 

The orders ID text should be 16px not 14px font. Right now it's smaller than everything else. 
Let's add a rule between the orders. The rule **should not** extend across the whole container. It should be the width of the order box. Here's a mock. 

![orders detail](https://user-images.githubusercontent.com/20409254/32246872-a3dac39c-be3d-11e7-9161-e330f4737f43.png)

efalayi commented:
@zenweasel at what screen size does this occur?

<img width="424" alt="iphone_x_-_ios_11_0_and_applications_and_cart_completed" src="https://user-images.githubusercontent.com/72819/32241510-91961bce-be2d-11e7-83a2-81a951ae1230.png">

I just tested with chrome tools.
